sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1329,18 +1329,19 @@ def test_vectorizer_stop_words_inconsistent():
             vec.fit_transform(["hello world"])
         # reset stop word validation
         del vec._stop_words_id
-        assert _check_stop_words_consistency(vec) is False
+        with pytest.warns(UserWarning, match=message):
```

Comment:
This is the actual fix, see previous discussion in https://github.com/scikit-learn/scikit-learn/pull/32023#discussion_r2309655370.

My guess right now is that the first call `_check_stop_words_consistency` was raising a warning and that the next `pytest.warns` was sometimes failing because of the default warnings `once` strategy. By putting this inside `pytest.warns` we avoid the side-effect and the warning is always issued.